### PR TITLE
atc: propagate span context into build containers

### LINKED ISSUE #5423: emit domain-specific spans from within tasks/resources
## What challenge are you facing?

I have a task that makes a request to an internal web service that my team has written. Both my Concourse cluster and this web service are emitting spans to the same Jaeger cluster, but I would like some way to link the spans from Concourse and the spans from my service under a common trace (the request being made inside the task container).

## What would make this better?

Since the w3c defines a standard for the `traceparent` header, let's try the following: if tracing is enabled, any task/resource get/resource put script should have a `TRACE_PARENT` env var set where the value is the trace ID of the build in question.

## Are you interested in implementing this yourself?

:+1:

### PR DESCRIPTION
**Pull request by aoldershaw**
~**https://github.com/concourse/concourse/pull/5666 should be merged first**~ - merged

## What does this PR accomplish?

Feature

closes #5423  .

## Changes proposed by this PR:

* adds a `TRACEPARENT` environment variable to all `task`, `get`, `put`, and `check` containers

## Notes to reviewer:

This is a simplified version of #5461 that only includes the `TRACEPARENT` propagation, and does not bump OpenTelemetry or introduce a Ginkgo OpenTelemetry reporter.

This can be manually tested by:

1. Running Concourse with the `hack/overrides/jaeger.yml` override
1. Running `fly execute` with the following task config: https://github.com/concourse/concourse/blob/199acfad85499ed82d4c3561687b60ffd6329a73/hack/tracing-task.yml

In Jaeger, you should see a child-span of the task step for the `downstream-service`

## Contributor Checklist
- [x] Followed [Code of conduct], [Contributing Guide] & avoided [Anti-patterns]
- [x] [Signed] all commits
- [x] Added tests (Unit and/or Integration)
- [ ] Updated [Documentation]
- [ ] Updated [Release notes]

[Code of Conduct]: https://github.com/concourse/concourse/blob/master/CODE_OF_CONDUCT.md
[Contributing Guide]: https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md
[Anti-patterns]: https://github.com/concourse/concourse/wiki/Anti-Patterns
[Signed]: https://help.github.com/en/github/authenticating-to-github/signing-commits
[Documentation]: https://github.com/concourse/docs
[Release notes]: https://github.com/concourse/concourse/tree/master/release-notes

## Reviewer Checklist
- [x] Code reviewed
- [x] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the
  [BOSH](https://github.com/concourse/concourse-bosh-release) and
  [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for
  the [integration
  tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env)
  (for example, if they are Garden configs that are not displayed in the
  `--help` text).

### CODE REVIEW

**Comment by aoldershaw** [comment]
Converting to a draft for now until we figure out what to do about check containers

**Comment by jomsie** [comment]
@aoldershaw merge conflict looks easy enough, just some imports?

**Comment by aoldershaw** [comment]
@jomsie yup, fixed

**Comment by jomsie** [comment]
> @jomsie yup, fixed

now DCO 😅 

anywho i'll start reviewin'!

**Comment by aoldershaw** [comment]
@jomsie added a release note. @jamieklassen if you have a chance, could you also look over the release note? I was struggling to eloquently describe the motivation for this change

File: release-notes/latest.md [hunk 1]
```diff
@@ -24,10 +24,21 @@ Currently the only API action that can be limited in this way is `ListAllJobs` -
 
 * When [distributed tracing](https://concourse-ci.org/tracing.html) is configured, Concourse will now emit spans for several of its backend operations, including resource scanning, check execution, and job scheduling. These spans will be appropriately linked when viewed in a tracing tool like Jaeger, allowing operators to better observe the events that occur between resource checking and build execution. #5659
 
+#### <sub><sup><a name="5653" href="#5653">:link:</a></sup></sub> feature
+
+* When [distributed tracing](https://concourse-ci.org/tracing.html) is configured, all check, get, put, and task containers will be run with the `TRACEPARENT` environment variable, which contains information of the parent span following the [w3c recommendation format](https://www.w3.org/TR/trace-context/#traceparent-header):
```
Comment (jamieklassen) [inline release-notes/latest.md:29]:
```suggestion
* When [distributed tracing](https://concourse-ci.org/tracing.html) is configured, all check, get, put, and task containers will be run with the `TRACEPARENT` environment variable set, which contains information about the parent span following the [w3c trace context format](https://www.w3.org/TR/trace-context/#traceparent-header):
```

**Pull request merged**